### ISSUE TITLE
Type-check body of IntEnum subclass better

### ISSUE THREAD

**Issue by gvanrossum** [python/mypy#1555]
This is a left-over task from PR #1535, in particular [this comment from Jukka](https://github.com/python/mypy/pull/1535#issuecomment-219790552):

> Here is an example of a thing that is still not quite right (no need to fix this now):

```python
class X(IntEnum):
    x = 'x'   # mypy doesn't complain
```
